### PR TITLE
Support AGT Counter driver for Renesas RA6, RA4, RA2 devices

### PR DESCRIPTION
**Pull request by khoa-nguyen-18**
Add support for AGT Counter driver for RA6, RA4, RA2 devices

- Add AGT Counter driver support on RA6M1, RA6M2, RA6M3, RA6M4, RA6M5, RA6E1, RA4M2, RA4M3, RA4W1, RA2A1
- Add support `samples/drivers/counter/alarm` for RA6M1, RA6M2, RA6M3, RA6M4, RA6M5, RA6E1, RA4M2, RA4M3, RA4W1, RA2A1

### CODE REVIEW

File: boards/renesas/ek_ra6m1/doc/index.rst [hunk 1]
```diff
@@ -94,6 +94,8 @@ The below features are currently supported on Zephyr OS for EK-RA6M1 board:
 +-----------+------------+----------------------+
 | CLOCK     | on-chip    | clock control        |
 +-----------+------------+----------------------+
+| COUNTER   | on-chip    | counter              |
```
Comment (soburi) [inline boards/renesas/ek_ra6m1/doc/index.rst:97]: Wrong commit message, should start with `boards: renesas:...`

File: dts/arm/renesas/ra/ra6/r7fa6e2bx.dtsi [hunk 2]
```diff
@@ -7,6 +7,13 @@
 #include <arm/renesas/ra/ra6/ra6-cm33-common.dtsi>
 #include <zephyr/dt-bindings/clock/ra_clock.h>
 
+/delete-node/ &agt0;
```
Comment (soburi) [inline dts/arm/renesas/ra/ra6/r7fa6e2bx.dtsi:10]: It's better to split the changes in the dts folder into a separate commit.

**Review by soburi — CHANGES_REQUESTED** [review]

File: boards/renesas/ek_ra6m1/doc/index.rst (hunk 1 above)
Comment (khoa-nguyen-18) [reply boards/renesas/ek_ra6m1/doc/index.rst:97]: Thank you, I have changed the commit message and split this modification into a commit

File: dts/arm/renesas/ra/ra6/r7fa6e2bx.dtsi (hunk 2 above)
Comment (khoa-nguyen-18) [reply dts/arm/renesas/ra/ra6/r7fa6e2bx.dtsi:10]: Thank you, I have split this dts modification into a commit. Could you please re-check again?

**Review by soburi — DISMISSED** [review]
LGTM

**Review by KhiemNguyenT — DISMISSED** [review]

**Comment by soburi** [comment]
@nordic-krch 
Could you take a look please?

**Review by nordic-krch — DISMISSED** [review]

**Comment by khoa-nguyen-18** [comment]
I just rebased main to solve the conflict. @soburi @KhiemNguyenT @nordic-krch Sorry for the inconvenience, can you please take a look?

**Review by nordic-krch — APPROVED** [review]

**Review by soburi — APPROVED** [review]

**Pull request merged**